include/deal.II/base/discrete_time.h

```diff
@@ -78,24 +87,178 @@ DEAL_II_NAMESPACE_OPEN
  * only a *desired* step size. You can query the actual time step size using the
  * get_next_step_size() function.
  *
+ *
+ * ### Details of time-stepping
+ *
+ * Since time is marched forward in a discrete manner in our simulations, we
+ * need to discuss how we increment time. During time stepping we enter two
+ * separate alternating regimes in every step.
+ *
+ * * The **snapshot** stage (the **current** stage, the **consistent**
+ *   stage): In this part of the algorithm, we are at $t = t_n$ and all
+ *   quantities of the simulation (displacements, strains, temperatures, etc.)
+ *   are up-to-date for $t = t_n$. In this stage, *current time* refers to
+ *   $t_n$, *next time* refers to $t_{n+1}$, *previous time* refers to
+ *   $t_{n-1}$. The other useful notation quantities are the *next* time step
+ *   size $t_{n+1} - t_n$ and *previous* time step size $t_n - t_{n-1}$. In
+ *   this stage, it is a perfect occasion to generate text output using print
+ *   commands within the user's code. Additionally, post-processed outputs can
+ *   be prepared here which can be viewed later by visualization programs such
+ *   as `Tecplot`, `Paraview`, and `VisIt`. Additionally, during the snapshot
+ *   stage, the code can assess the quality of the previous step and decide
+ *   whether it wants to increase or decrease the time step size. The step
+ *   size for the next time step can be modified here.
+ * * The **update** stage (the **transition** stage, the **inconsistent**
+ *   stage): In this section of the program, the internal state of the
+ *   simulation is getting updated from $t_n$ to $t_{n+1}$. All of the
+ *   variables need to be updated one by one, the step number is incremented,
+ *   the time is incremented by $dt = t_{n+1} - t_n$, and time-integration
+ *   algorithms are used to update the other simulation quantities. In the
+ *   middle of this stage, some variables have been updated to $t_{n+1}$ but
+ *   other variables still represent their value at $t_n$. Thus, we call this
+ *   the inconsistent stage, requiring that no post-processing output related
+ *   to the state variables take place within it. The state variables, namely
+ *   those related to time, the solution field and any internal variables, are
+ *   not synchronized and the get updated one by one. In general, the order of
+ *   updating variables is arbitrary, but some care should be taken if there
+ *   are interdependencies between them. For example, if some variable such as
+ *   $x$ depends on the calculation of another variable such as $y$, then $y$
+ *   must be updated before $x$ can be updated. The question arises whether
```

Comment:
I'd make that a paragraph break, but don't know how that would look in the documentation -- is it still the same bullet point?
```suggestion
 *   must be updated before $x$ can be updated.
 * 
 *   The question arises whether
```